eZ/Bundle/EzPublishCoreBundle/Imagine/ResolverFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface;
+
+class ResolverFactory
+{
+    /**
+     * @var \Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface
+     */
+    private $resolver;
+
+    /**
+     * @var string
+     */
+    private $proxyResolverClass;
+
+    /**
+     * @var array
+     */
+    private $hosts = [];
+
+    public function __construct(ConfigResolverInterface $configResolver, ResolverInterface $resolver, $proxyResolverClass)
+    {
+        $this->resolver = $resolver;
+        $this->proxyResolverClass = $proxyResolverClass;
+
+        if ($configResolver->hasParameter('image_host') &&
+            ($imageHost = $configResolver->getParameter('image_host')) !== '') {
+            $this->hosts = [$imageHost];
```

Comment:
Looked a bit deeper, and from what I can see we are basically hitting this part of the code right?
https://github.com/liip/LiipImagineBundle/blob/master/Imagine/Cache/Resolver/ProxyResolver.php#L90

It has inline comment "BC".
However looking at blame, it does not look to be deprecated so +1 from my side but thought I would mention it.

Reply:
Yes, exactly, we are using that part of the code.

As for me this `BC` comment is due to the fact, that when they changed `Resolver` behaviour, it was kind of BC break, please take a look: https://github.com/liip/LiipImagineBundle/pull/687/files#diff-f1e2882ffa991d3494a674108aa8000f

Reply:
yes